docs/testing/03-sdk-tests.mdx

```diff
@@ -0,0 +1,213 @@
+---
+sidebar_position: 3
+sidebar_label: "SDK Tests"
+id: kuneiform-sdk-tests
+title: SDK tests
+description: Testing Kuneiform with the testing SDK
+slug: /testing/go-sdk
+---
+
+To use the testing SDK, you will need to have Go installed and have a project set up.
+This document assumes that you are generally familiar with Go and the standard library testing.
+For more information on how to set this up, please refer to the [Go Documentation](<https://go.dev/doc/tutorial/add-a-test>).
+
+## Installation
+
+To install the testing SDK, run:
+
+```shell
+$ go get github.com/kwilteam/kwil-db/testing
+```
+
+## Making The Test
+
+To make a test, create a new test file and import the testing package:
+
+```go title="main_test.go"
+package main_test
+
+import (
+	"testing"
+
+	kwilTesting "github.com/kwilteam/kwil-db/testing"
+)
+
+func Test_Kuneiform(t *testing.T) {
+	kwilTesting.RunSchemaTest(t, kwilTesting.SchemaTest{})
+}
+```
+
+Below is a full reference for the `kwilTesting.SchemaTest` struct:
+
+<details>
+  <summary> Kwil Go Testing Reference </summary>
+
+```go
+// SchemaTest allows for testing schemas against a live database.
+// It allows for several ways of specifying schemas to deploy, as well
+// as functions that can be run against the schemas, and expected results.
+type SchemaTest struct {
+	// Name is the name of the test case.
+	Name string `json:"name"`
+	// Schemas are plain text schemas to deploy as
+	// part of the text.
+	Schemas []string `json:"-"`
+	// SchemaFiles are paths to the schema files to deploy.
+	SchemaFiles []string `json:"schema_files"`
+	// SeedStatements are SQL statements run before each test that are
+	// meant to seed the database with data. It maps the database name
+	// to the SQL statements to run. The name is the database name,
+	// defined using "database <name>;". The test case will derive the
+	// DBID from the name.
+	SeedStatements map[string][]string `json:"seed_statements"`
+	// TestCases execute actions or procedures against the database
+	// engine, taking certain inputs and expecting certain outputs or
+	// errors. These run separately from the functions, and separately
+	// from each other. They are the easiest way to test the database
+	// engine, but if more nuanced tests are needed (e.g. to simulate
+	// several different wallets), the FunctionTests field should be used
+	// instead. All schemas will be redeployed and all seed data re-applied
+	// between executing each TestCase.
+	TestCases []TestCase `json:"test_cases"`
+	// FunctionTests are arbitrary functions that can be used to
+	// execute any logic against the schemas.
+	// All schemas will be reset before each function is run.
+	// FunctionTests are more cumbersome to use than TestCases, but
+	// they allow for more nuanced testing and flexibility.
+	// All functions and testcases are run against fresh schemas.
+	FunctionTests []TestFunc `json:"-"`
```

Comment:
Should you add an example below for `FunctionTests`?

Reply:
Yeah probably. Will add that.